### PR TITLE
Add IntersectionObserver sample

### PR DESCRIPTION
**Pull request by surma**
I don’t know how to build the sample locally so I am not sure I did it right.

@gauntface @PaulKinlan @jeffposnick Any of you mind giving it a thorough review?

### CODE REVIEW

File: intersection-observer/index.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  provides an API to understand the visibility and position of DOM elements relative to
+  a containing element or to the top-level viewport.
+</p>
+<p>
+  This sample illustrates using <code>IntersectionObserver</code> to implement a infinite
+  scroller without having to rely on scroll events. A sentinel element triggers the loading
+  of additional elements once it comes into view and is being recycled after the new
+  elements have been attached to the list.
+<p>
+  The <a href="https://developers.google.com/web/updates/???"><em>Google Devloper Web Updates</em></a>
```
Comment (beaufortfrancois) [inline intersection-observer/index.html]: Nit: `s/Devloper/Developer`

File: intersection-observer/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<!-- TODO: Replace PLACEHOLDER with feature name. -->
+PLACEHOLDER Sample
```
Comment (jeffposnick) [inline intersection-observer/README.md]: "PLACEHOLDER" → IntersectionObserver

File: intersection-observer/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+===
+<!-- TODO: Replace PLACEHOLDER in the path to correspond to the real github.io URL. -->
```
Comment (jeffposnick) [inline intersection-observer/README.md]: Remove comment.

File: intersection-observer/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+See https://googlechrome.github.io/samples/PLACEHOLDER/index.html for a live demo.
```
Comment (jeffposnick) [inline intersection-observer/README.md]: "PLACEHOLDER" → intersection-observer

File: intersection-observer/README.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+
+<!-- TODO: Replace PLACEHOLDER with the id from the chromestatus.com URL. -->
```
Comment (jeffposnick) [inline intersection-observer/README.md]: Remove comment.

File: intersection-observer/README.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+Learn more at https://www.chromestatus.com/feature/PLACEHOLDER
```
Comment (jeffposnick) [inline intersection-observer/README.md]: "PLACEHOLDER" → 5695342691483648

File: intersection-observer/index.html [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  provides an API to understand the visibility and position of DOM elements relative to
+  a containing element or to the top-level viewport.
+</p>
+<p>
+  This sample illustrates using <code>IntersectionObserver</code> to implement a infinite
+  scroller without having to rely on scroll events. A sentinel element triggers the loading
+  of additional elements once it comes into view and is being recycled after the new
+  elements have been attached to the list.
+<p>
+  The <a href="https://developers.google.com/web/updates/???"><em>Google Devlopers Web Updates</em></a>
```
Comment (jeffposnick) [inline intersection-observer/index.html]: As the article's author, you get to choose the final part of the URL's path, and the prefix is going to be https://developers.google.com/web/updates/2016/04/. You should be able to put in the actual value here, and we can hold off on merging until your article is published, if you'd like.

**Comment by jeffposnick** [comment]
Hey @surma—Sorry, the templating process is not as straightforward as it could be. What specifically went wrong when you tried to run a local Jekyll build? I'm happy to help debug.

In the meantime, I've take your code and got it working with our samples style and general setup. I have a copy deployed at https://jeffy.info/samples/intersection-observer/ and the changes I made are at https://github.com/GoogleChrome/samples/compare/intersectionobserver...jeffposnick:intersectionobserver

One thing that I wasn't sure about when testing out the code was that the intersection observer seems to be triggered twice each time you scroll down far enough. I'm not sure if that's due to the modifications I made or whether your original sample had that behavior as well.

File: intersection-observer/README.md (hunk 2 above)
Comment (surma) [inline intersection-observer/README.md]: Ugh. I forgot about the README. Sorry.

**Comment by surma** [comment]
Thanks a lot for your help, @jeffposnick!

I got Jekyll working now. I think my Ruby was just busted. I fixed the double call to the IO (working as intended – my bad) and filled out the README. PTAL :)

**Comment by jeffposnick** [comment]
Hey @surma—There are some linting issues identified by our CI server: https://travis-ci.org/GoogleChrome/samples/builds/123936454#L268

You can lint locally using `npm run lint` to confirm that it's good before pushing the revisions.

**Comment by jeffposnick** [comment]
👍 other than the linting fixes.

**Comment by surma** [comment]
@jeffposnick Linting has been fixed. Had to add a ESLint `global` comment for `IntersectionObserver`. If that is fine with you, feel free to merge at will :)

**Comment by jeffposnick** [comment]
👍  Thanks, Surma!

**Pull request merged**